Automattic/wp-calypso
# Tailored Flows: Use stepper's domain and plans for NL flow

## Proposed Changes

Newsletter flow depends on /start for domains and plans step, last year team Vertex moved [domains and plans into stepper](pdDR7T-nj-p2) and initially applied this change only on Link in Bio flow. This PR applies it on Newsletter flow.



## Testing Instructions

- Pull and run this branch or use calypso live links
- Navigate to `/setup/newsletter/intro`
- Complete the Newsletter flow
- Note that, on Domains and Plans step, the user should never be redirect to `/start` besides for login IF logged out
	- Domains step URL should be similar to `/setup/newsletter/domains...`  and not `start/newsletter/domains....`
	- Plans step URL should be similar to `/setup/newsletter/plans...` and not `start/newsletter/plans-newsletter...`

- Some cases to double check: 
	- Logged out user 
	- Free plan, free domain and/or both

- Bonus test ❤️  :
 	- Test Link In Bio flow should be intact.


Fixes #74378
